File: Java/Hard/WrongAnswer/reachable-nodes-in-subdivided-graph_9.java
class Solution {
    public int reachableNodes(int[][] edges, int maxMoves, int n) {
        
        int[][] graph = new int[n][n];
        for(int i=0;i<edges.length;i++){
            graph[edges[i][0]][edges[i][1]] = edges[i][2];
            graph[edges[i][1]][edges[i][0]] = edges[i][2];
        }
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b)->b[1]-a[1]);
        pq.offer(new int[]{0,maxMoves});
        boolean[] visited = new boolean[n];
        int res = 0;
        while(!pq.isEmpty()){
            int[] cur = pq.poll();
            if(visited[cur[0]]) continue;
            visited[cur[0]] = true;
            res++;
            for(int i=0;i<n;i++){
                if(graph[cur[0]][i]==0) continue;
                if(cur[1]>graph[cur[0]][i] && !visited[i]){
                    pq.offer(new int[]{i,cur[1]-graph[cur[0]][i]-1});
                }
                graph[i][cur[0]] -= Math.min(cur[1], graph[cur[0]][i]);
                res += Math.min(cur[1], graph[cur[0]][i]);
            }
        }
        return res;
    }
}

File: C++/Hard/Accepted/reachable-nodes-in-subdivided-graph_9.cpp
class Solution {
public:
    int reachableNodes(vector<vector<int>>& edges, int maxMoves, int n) {
        
        vector<vector<pair<int,int>>> graph(n);
        for(auto& e: edges) {
            graph[e[0]].push_back({e[1], e[2]});
            graph[e[1]].push_back({e[0], e[2]});
        }
        priority_queue<pair<int,int>> pq;
        pq.push({maxMoves, 0});
        vector<int> dist(n, -1);
        while(!pq.empty()) {
            int d = pq.top().first;
            int u = pq.top().second;
            pq.pop();
            if(dist[u] != -1) continue;
            dist[u] = d;
            for(auto& p: graph[u]) {
                int v = p.first;
                int w = p.second;
                if(dist[v] == -1 && d - w - 1 >= 0) {
                    pq.push({d - w - 1, v});
                }
            }
        }
        int ans = 0;
        for(int i = 0; i < n; ++i) ans += dist[i] >= 0;
        for(auto& e: edges) {
            int u = e[0];
            int v = e[1];
            int w = e[2];
            int d1 = dist[u] == -1 ? 0 : dist[u];
            int d2 = dist[v] == -1 ? 0 : dist[v];
            ans += min(d1 + d2, w);
        }
        return ans;
    }
};